experimental/ActionPolyRing/src/Operators.jl

```diff
@@ -0,0 +1,292 @@
+###############################################################################
+#
+#  Unary Operators 
+#
+###############################################################################
+
+Base.:-(apre::ActionPolyRingElem) = parent(apre)(-data(apre))
+
+###############################################################################
+#
+#  Binary Operators 
+#
+###############################################################################
+
+function Base.:+(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}) where {T<:RingElement}
+  check_parent(apre1, apre2)
+  return parent(apre1)(data(apre1) + data(apre2))
+end
+
+function Base.:-(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}) where {T<:RingElement}
+  check_parent(apre1, apre2)
+  return parent(apre1)(data(apre1) - data(apre2))
+end
+
+function Base.:*(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}) where {T<:RingElement}
+  check_parent(apre1, apre2)
+  return parent(apre1)(data(apre1) * data(apre2))
+end
+
+function Base.:/(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}) where {T<:RingElement}
+  check_parent(apre1, apre2)
+  return parent(apre1)(data(apre1) / data(apre2))
+end
+
+function divexact(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}; check::Bool = true) where {T<:RingElement}
+  check_parent(apre1, apre2)
+  return parent(apre1)(divexact(data(apre1), data(apre2)))
+end
+
+function mod(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}) where {T<:RingElement}
+  check_parent(apre1, apre2)
+  return parent(apre1)(mod(data(apre1), data(apre2)))
+end
+
+function gcd(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}) where {T<:RingElement}
+  check_parent(apre1, apre2) 
+  return parent(apre1)(gcd(data(apre1), data(apre2)))
+end
+
+function lcm(apre1::ActionPolyRingElem{T}, apre2::ActionPolyRingElem{T}) where {T<:RingElement}
+   check_parent(apre1, apre2)
+   return parent(apre1)(lcm(data(apre1), data(apre2)))
+end
+
+###############################################################################
+#
+#  Comparison 
+#
+###############################################################################
+
+### Difference ###
+Base.:(==)(dpre1::DifferencePolyRingElem{T}, dpre2::DifferencePolyRingElem{T}) where {T<:RingElement} = data(dpre1) == data(dpre2) && parent(dpre1) === parent(dpre2)
```

Comment:
```suggestion
Base.:(==)(dpre1::DifferencePolyRingElem{T}, dpre2::DifferencePolyRingElem{T}) where {T<:RingElement} = parent(dpre1) === parent(dpre2) && data(dpre1) == data(dpre2)
```
comparing the parents is basically instantaneous and this way you take the shortcut faster (if the first arg of `&&` is false, the second one is skipped)